packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.controller.js

```diff
@@ -21,4 +21,23 @@ export default class PublicGatewaysController {
   refreshMessages() {
     this.messages = this.messageHandler.getMessages();
   }
+
+  onAddPublicGatewayClick() {
+    this.trackPublicGateways('add');
+    this.goToAddPublicGateway();
```

Comment:
**question:** Where is `goToAddPublicGateway` function defined?

Reply:
@anooparveti  goToAddPublicGateway is there in add components, which is currently in block state. There we have this method, I have added this so that I don't need to make any change in here.